File: python/src/qdk_chemistry/algorithms/symmetry_shift.py
"""Public entry point for the symmetry shift algorithm.

This module re-exports the abstract :class:`SymmetryShifter` interface, the
:class:`FermionicLowRankShifter` implementation, and the :class:`SymmetryShift`
parameter container so that consumers can import them directly from ``qdk_chemistry.algorithms`` without
depending on internal package paths.
"""

# --------------------------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See LICENSE.txt in the project root for license information.
# --------------------------------------------------------------------------------------------

from qdk_chemistry._core._algorithms import (
    FermionicLowRankShifter,  # noqa: F401 - re-export
    SymmetryShift,  # noqa: F401 - re-export
    SymmetryShifter,  # noqa: F401 - re-export
)

File: cpp/src/qdk/chemistry/algorithms/symmetry_shift.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE.txt in the project root for
// license information.

#include <memory>
#include <qdk/chemistry/algorithms/symmetry_shift.hpp>
#include <qdk/chemistry/utils/logger.hpp>
#include <string>

#include "microsoft/symmetry_shift/fermionic_low_rank.hpp"

namespace qdk::chemistry::algorithms {

// ---------------------------------------------------------------------------
// Factory registration.
// ---------------------------------------------------------------------------

namespace {

std::unique_ptr<SymmetryShifter> make_fermionic_low_rank_shifter() {
  QDK_LOG_TRACE_ENTERING();
  return std::make_unique<microsoft::FermionicLowRankShifter>();
}

}  // namespace

void SymmetryShifterFactory::register_default_instances() {
  QDK_LOG_TRACE_ENTERING();

  SymmetryShifterFactory::register_instance(&make_fermionic_low_rank_shifter);
}

}  // namespace qdk::chemistry::algorithms
